packages/db/tests/translate/test-granules.js

```diff
@@ -789,6 +789,237 @@ test('translateApiGranuleToPostgresGranuleWithoutNilsRemoved does not remove nul
+  const pdrCumulusId = 4;
+  const granuleStatus = 'complete';
+
+  const fakeCollectionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(collectionCumulusId),
+  };
+  const fakeProviderPgModel = {
+    getRecordCumulusId: () => Promise.resolve(providerCumulusId),
+  };
+  const fakePdrPgModel = {
+    getRecordCumulusId: () => Promise.resolve(pdrCumulusId),
+  };
+
+  const apiGranule = {
+    granuleId: cryptoRandomString({ length: 5 }),
+    collectionId: null,
+    status: granuleStatus,
+  };
+
+  await t.throwsAsync(translateApiGranuleToPostgresGranuleWithoutNilsRemoved({
+    dynamoRecord: apiGranule,
+    knexOrTransaction: {},
+    collectionPgModel: fakeCollectionPgModel,
+    pdrPgModel: fakePdrPgModel,
+    providerPgModel: fakeProviderPgModel,
+  }), { instanceOf: ValidationError });
+});
+
+test.serial('translateApiGranuleToPostgresGranuleWithoutNilsRemoved throws on inappropriate nullification of granuleId', async (t) => {
+  const collectionCumulusId = 1;
+  const providerCumulusId = 2;
+  const pdrCumulusId = 4;
+  const granuleStatus = 'complete';
+
+  const fakeCollectionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(collectionCumulusId),
+  };
+  const fakeProviderPgModel = {
+    getRecordCumulusId: () => Promise.resolve(providerCumulusId),
+  };
+  const fakePdrPgModel = {
+    getRecordCumulusId: () => Promise.resolve(pdrCumulusId),
+  };
+
+  const apiGranule = {
+    granuleId: null,
+    collectionId: collectionCumulusId,
+    status: granuleStatus,
+  };
+
+  await t.throwsAsync(translateApiGranuleToPostgresGranuleWithoutNilsRemoved({
+    dynamoRecord: apiGranule,
+    knexOrTransaction: {},
+    collectionPgModel: fakeCollectionPgModel,
+    pdrPgModel: fakePdrPgModel,
+    providerPgModel: fakeProviderPgModel,
+  }), { instanceOf: ValidationError });
+});
+
+test.serial('translateApiGranuleToPostgresGranuleWithoutNilsRemoved throws on inappropriate nullification of status', async (t) => {
+  const collectionCumulusId = 1;
+  const providerCumulusId = 2;
+  const pdrCumulusId = 4;
+
+  const fakeCollectionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(collectionCumulusId),
+  };
+  const fakeProviderPgModel = {
+    getRecordCumulusId: () => Promise.resolve(providerCumulusId),
+  };
+  const fakePdrPgModel = {
+    getRecordCumulusId: () => Promise.resolve(pdrCumulusId),
+  };
+
+  const apiGranule = {
+    granuleId: cryptoRandomString({ length: 5 }),
+    collectionId: collectionCumulusId,
+    status: null,
+  };
+
+  await t.throwsAsync(translateApiGranuleToPostgresGranuleWithoutNilsRemoved({
+    dynamoRecord: apiGranule,
+    knexOrTransaction: {},
+    collectionPgModel: fakeCollectionPgModel,
+    pdrPgModel: fakePdrPgModel,
+    providerPgModel: fakeProviderPgModel,
+  }), { instanceOf: ValidationError });
+});
+
+test.serial('translateApiGranuleToPostgresGranuleWithoutNilsRemoved converts API granule to Postgres, preserving null values', async (t) => {
```

Comment:
The `undefined` values are also preserved

Reply:
5a1f500f08